### PR TITLE
Add ranking score threshold

### LINKED ISSUE #753: [v1.9] Support `rankingScoreThreshold` search parameter
*Following this [central issue](https://github.com/meilisearch/integration-guides/issues/301)*

This is related to a newly introduced feature in [Meilisearch v1.9.0](https://github.com/meilisearch/meilisearch/releases/tag/v1.9.0): the `rankingScoreThreshold` search parameter.

Refer to [docs for more information](https://www.meilisearch.com/docs/reference/api/search#ranking-score-threshold).

- [ ] Ensure the library can accept the new `rankingScoreThreshold` search parameter.
- [ ] ⚠️ Add integration tests ⚠️
- [ ] Update the `.code-samples.meilisearch.yaml`
    - [ ]  Add `search_parameter_reference_ranking_score_threshold_1` key and “translate” the following curl example by using the newly added methods: https://github.com/meilisearch/documentation/blob/04f26a35771305378ed0a506f4d60ef06d07bd7e/.code-samples.meilisearch.yaml#L1344

### PR DESCRIPTION
**Pull request by the-sinner**
# Pull Request

## Related issue
Fixes #753 

## What does this PR do?
- Add ranking score threshold

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -417,6 +417,9 @@ search_parameter_guide_show_ranking_score_1: |-
 search_parameter_guide_show_ranking_score_details_1: |-
   SearchRequest searchRequest = SearchRequest.builder().q("dragon").showRankingScoreDetails(true).build();
   client.index("movies").search(searchRequest);
+search_parameter_reference_ranking_score_threshold_1: |-
+  SearchRequest searchRequest = SearchRequest.builder().q("badman").rankingScoreThreshold(0.2).build();
+  client.index("movies").search(searchRequest);
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:422]:
```suggestion
  client.index("INDEX_NAME").search(searchRequest);
```

File: src/main/java/com/meilisearch/sdk/Search.java [hunk 2]
```diff
@@ -70,6 +70,12 @@ SearchResult search(String uid, String q) throws MeilisearchException {
      * @throws MeilisearchException Search Exception or Client Error
      */
     Searchable search(String uid, SearchRequest sr) throws MeilisearchException {
+        if (sr != null
+                && sr.rankingScoreThreshold != null
+                && (sr.rankingScoreThreshold > 1.0 || sr.rankingScoreThreshold < 0)) {
+            throw new MeilisearchException(
+                    "Ranking score threshold must be between 0 and 1.0 inclusive");
+        }
```
Comment (curquiza) [inline src/main/java/com/meilisearch/sdk/Search.java:78]: No need to handle the error here, Meilisearch will return the appropriate error

**Review by curquiza — CHANGES_REQUESTED** [review]
Thank you @the-sinner 🙏

File: src/main/java/com/meilisearch/sdk/Search.java (hunk 2 above)
Comment (the-sinner) [reply src/main/java/com/meilisearch/sdk/Search.java]: done

**Review by curquiza — APPROVED** [review]
Niiiice

bors merge

**Pull request merged**